ext/background/suggestions.js

```diff
@@ -1,14 +1,60 @@
-let _recentSuggestions = [];
+let _lastFetch = 0;
+let _attachments = [];
 
-// Returns an array of recent attachments that have been suggested via the
-// extension.
-function recentSuggestions() {
-  return Promise.resolve(_recentSuggestions);
+// Returns an array of attachments that have been suggested via the extension.
+function getAttachments() {
+  if (Date.now() - _lastFetch < 1000 * 60) {
+    return Promise.resolve(_attachments);
+  }
+  return loadAttachments()
+    .then(() => _attachments)
+    .catch(e => {
+      console.error('Failed to load attachments: ', e);
+      return _attachments;
+    });
+}
+
+// Sort, remove duplicates via source_id, and keep track of only last 100 suggestions.
+function cleanAttachments() {
+  let seen = {};
+  _attachments.forEach(
+    a => (a.display_date = a.display_date || a.date_modified || a.date_created || a.created_at)
+  );
+  _attachments.sort((a, b) => new Date(b.display_date) - new Date(a.display_date));
+  _attachments = _attachments.filter(attachment =>
+    seen[attachment.source_id] ? false : (seen[attachment.source_id] = 1)
+  );
+  _attachments.length = Math.min(_attachments.length, 100);
+}
+
+function storeAttachments() {
+  chrome.storage.local.set({ attachments: _attachments });
+}
+
+function loadAttachments() {
+  let after = new Date(Date.now() - 24 * 60 * 60 * 1000).toISOString();
+  return orgs()
+    .then(slugs => Promise.all(slugs.map(getSession)))
+    .then(sessions =>
+      Promise.all(
+        sessions.map(s => {
+          return listAttachments(s.user.user_id, after, authorize(s)).then(resp => {
+            console.log(resp);
+            resp.attachments.forEach(attachment => _attachments.push(attachment));
+          });
+        })
+      )
+    )
+    .then(() => {
+      cleanAttachments();
+      storeAttachments();
+      _lastFetch = Date.now();
+    });
 }
 
 // Background page is ephemeral, so load stored version of suggestions.
-chrome.storage.local.get('suggestions', function(result) {
-  _recentSuggestions = result.suggestions || [];
+chrome.storage.local.get('attachments', function(result) {
+  _attachments = result.suggestions || [];
```

Comment:
think this needs to be `= result.attachments` per line 31's `set({attachments: _attachments})`

Reply:
Good catch.